reflectapi-demo/src/tests/serde.rs

```diff
@@ -110,7 +110,8 @@ fn test_enum_untagged() {
 // see https://github.com/thepartly/reflectapi/issues/1
 enum TestEnumTag {
     Variant1 { field_name: u8 },
-    Variant2(u8),
+    Variant2 { field_name: u8 },
+    // Variant2(u8),
```

Comment:
why do we need to change test code?

Reply:
It hits an assertion I put in. There's no point allowing this since serde-json will panic if we try to use it. https://github.com/thepartly/reflectapi/issues/1

Reply:
I can relax the assertion if you prefer.
```
                match &mut strukt.fields {
                    reflectapi_schema::Fields::Named(fields) => fields.push(tag_field),
                    reflectapi_schema::Fields::None => {
                        strukt.fields = reflectapi_schema::Fields::Named(vec![tag_field])
                    }
                    reflectapi_schema::Fields::Unnamed(_) => {
                        panic!("internal repr with unnamed fields is not allowed")
                    }
                }
```

Reply:
Ahh... are you planning to address it later? I think this code sample needs to move to the test suite where we test compilation error messages.

Reply:
I can try along with the other compile error. I'll discuss offline. The problem is I imagine serde would have implemented the compile time error already if it were easy.

Reply:
I think serde just missed it. It is not difficult to implement.

Reply:
ok, merge this and address denying at compile time in the separate MR